### PR TITLE
Move compute docs out

### LINKED ISSUE #488: Move compute requirements docs
Somewhat related to #458 (in that this is the other doc in that current section), we're also going to move the `compute-requirements.md` doc into `contributing-to-analyses/analysis-modules` between the running and documenting docs. 

At this point, the **Determining compute requirements** section will no longer be needed in the contributing to analysis docs.

### PR DESCRIPTION
**Pull request by sjspielman**
Closes #488 

This PR moves out the compute requirement docs into `analysis-modules`, and updates relative links accordingly.

### CODE REVIEW

**Comment by sjspielman** [comment]
Noting that _deleting_ the entire `determining-requirements` directory is part of #513, since that change comes with a bunch more link overhauls!

File: docs/contributing-to-analyses/analysis-modules/compute-requirements.md [hunk 1]
```diff
@@ -1,7 +1,12 @@
...
+# Module compute requirements
 
-We use the term **compute requirements** to refer to the computational resources, such as the number of CPUs or the amount of RAM, that are required for your analysis to run.
-Reporting the compute requirements for running your analysis to project organizers helps us plan for allocating resources on Amazon Web Services (AWS) and to determine how to [review your pull requests](../pr-review-and-merge/index.md) (e.g., using our laptops vs. using AWS). <!-- STUB_LINK: may replace with review-specific page -->
+When planning and implementing your analysis module, you will need to consider your module's **compute requirements.**
+This refers to the computational resources, such as the number of CPUs or the amount of RAM, that are required for your analysis to run.
+
+Knowing and reporting this information helps the OpenScPCA project organizers:
+
+- It helps us plan and allocate resources on Amazon Web Services (AWS) to run your module
```
Comment (jaclyn-taroni) [inline docs/contributing-to-analyses/analysis-modules/compute-requirements.md:8]:
```suggestion
- It helps us plan and allocate resources on Amazon Web Services (AWS) to run your module.
```

File: docs/contributing-to-analyses/analysis-modules/compute-requirements.md [hunk 2]
```diff
@@ -1,7 +1,12 @@
...
+- It provides the compute context we need while [reviewing your pull requests](../pr-review-and-merge/index.md) (e.g., can we review using our laptops vs. using AWS).
```
Comment (jaclyn-taroni) [inline docs/contributing-to-analyses/analysis-modules/compute-requirements.md:9]: "provides the compute context we need" seemed unclear

```suggestion
- It lets us know what compute resources we need while [reviewing your pull requests](../pr-review-and-merge/index.md) (e.g., can we review using our laptops vs. using AWS).
```

File: docs/contributing-to-analyses/determining-requirements/index.md [hunk 3]
```diff
@@ -6,8 +6,3 @@ When performing an analysis, there are two sets of requirements to be aware of a
 Tracking and reporting dependencies is important so that others can run your code and reproduce your results.
 
     - See the [Software Requirements](./determining-software-requirements.md) page for approaches to track and manage dependencies.
-
-1. **Compute requirements** refer to the computational resources, such as the number of CPUs or the amount of RAM, that are required for your analysis to run.
```
Comment (jaclyn-taroni) [inline docs/contributing-to-analyses/determining-requirements/index.md:10]: Shouldn't the rest of this document change then, too? For example, there are no longer two sets of requirements being explained here.

**Review by jaclyn-taroni — COMMENTED** [review]
Most of this looks good, but I think `docs/contributing-to-analyses/determining-requirements/index.md` needs another pass before I can approve.

**Comment by sjspielman** [comment]
> Most of this looks good, but I think docs/contributing-to-analyses/determining-requirements/index.md needs another pass before I can approve.

Sorry if I was not clear about this - This file is actually being _deleted_ in #513, along with the entire `determining-requirements` directory. It will not exist anymore after moving all software and compute docs out.

**Review by jaclyn-taroni — APPROVED** [review]
👍🏻 assuming you take my other suggestion

This information belonged in the initial post, in my opinion: https://github.com/AlexsLemonade/OpenScPCA-analysis/pull/516#issuecomment-2161150507

**Comment by sjspielman** [comment]
It was mostly here, but could have been there too! https://github.com/AlexsLemonade/OpenScPCA-analysis/pull/516#issuecomment-2161150507

**Pull request merged**